newIDE/app/src/AssetStore/CreditsPackages/CreditsPackagePurchaseDialog.js

```diff
@@ -301,7 +309,13 @@ const CreditsPackagePurchaseDialog = ({
     <FlatButton
       key="cancel"
       label={purchaseSuccessful ? <Trans>Close</Trans> : <Trans>Cancel</Trans>}
-      onClick={onClose}
+      onClick={() => {
+        if (purchaseSuccessful) {
+          onPurchaseSuccessful();
+        }
+
+        onClose();
```

Comment:
Should this function be called `onCancel` to match the button label?
Or is it `onPurchaseSuccessful` that should be renamed `onCloseWhenPurchaseSuccessful`

Reply:
(I feel like the naming is a bit confusing)

Reply:
Hmm, actually it's not a if/else, it's just that we use one button to close.
If the purchase was successful, we also call the `onPurchaseSuccessful` callback

Reply:
That's why I feel it's confusing: the trigger to call the callback is not the "purchase successful" event.

Reply:
I've updated the name 👍